### ISSUE TITLE
Error running test code `AttributeError: type object 'TranslationDataset' has no attribute 'name'`

### ISSUE THREAD

**Issue by Rj7** [pytorch/text#233]
When I try to load data a translation dataset using `TranslationDataset` class I get this error. There is no attribute named `name` in `Dataset` class or `TranslationDataset`. 

I tried do as specified in one of the test cases. https://github.com/pytorch/text/blob/master/test/translation.py#L75

`data_set = datasets.TranslationDataset.splits(root='../data/',train='training.10k', validation='newstest2012', test='newstest2015',exts=('.de', '.en'), fields=(DE, EN))`

```
~/anaconda3/envs/opennmt/lib/python3.6/site-packages/torchtext/datasets/translation.py in splits(cls, exts, fields, root, train, validation, test, **kwargs)
     57                 Dataset.
     58         """
---> 59         path = cls.download(root)
     60 
     61         train_data = None if train is None else cls(

~/anaconda3/envs/opennmt/lib/python3.6/site-packages/torchtext/data/dataset.py in download(cls, root, check)
    109             dataset_path (str): Path to extracted dataset.
    110         """
--> 111         path = os.path.join(root, cls.name)
    112         check = path if check is None else check
    113         if not os.path.isdir(check):

AttributeError: type object 'TranslationDataset' has no attribute 'name'
```

**Comment by mttk** [comment]
@Rj7 the `TranslationDataset` class is not an actual dataset, and not passing the `path` argument (which loads the dataset) you are forcing the dataset to download itself (which it can't, because it's just a superclass for the other datasets in `torchtext.datasets.translation.py`).

This works in the test code because in line https://github.com/pytorch/text/blob/master/test/translation.py#L52 the `Multi30k` dataset is downloaded first, and then the path it was downloaded to (default: `'.data/multi30k'`) is passed as an argument to the `TranslationDataset` in line https://github.com/pytorch/text/blob/master/test/translation.py#L80

To summarize: 
- if you want to use the superclass `TranslationDataset` you need to pass the `path` argument to the location of an already downloaded dataset
- otherwise, use a concrete implementation `Multi30k` without `path` and the dataset will be downloaded. 

Hope this helps.

**Comment by Rj7** [comment]
Awesome. It makes sense now. Thank you!
